phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -265,12 +332,18 @@ public SingleTableColumnResolver(PhoenixConnection connection, NamedTableNode ta
                if (def.getColumnDefName().getFamilyName() != null) {
                    families.add(new PColumnFamilyImpl(PNameFactory.newName(def.getColumnDefName().getFamilyName()),Collections.<PColumn>emptyList()));
                }
-           }
-           Long scn = connection.getSCN();
-           PTable theTable = new PTableImpl(connection.getTenantId(), table.getName().getSchemaName(), table.getName().getTableName(), scn == null ? HConstants.LATEST_TIMESTAMP : scn, families);
+            }
+            Long scn = connection.getSCN();
+            String schema = table.getName().getSchemaName();
+            if (connection.getSchema() != null) {
```

Comment:
Not sure if functionally this is the right thing to do here. I would expect Phoenix to thrown an exception when the connection's setting of schema/namespace is different from the the table's namespace/schema. Or is my understanding not correct here? Would be good to add a test case for this too in UseSchemaIT.